src/main/scala/uoption/package.scala

```diff
@@ -1,11 +1,11 @@
 import scala.language.higherKinds
 
 package object uoption {
-  private[uoption] final class WrappedNone(
+  private[uoption] sealed class WrappedNone(
       depth: Int,
-      val unwrap: Any // UNone or WrappedNone
+      private[uoption] val unwrap: WrappedNone // null when this is UNone
   ) {
-    lazy val wrap: WrappedNone =
+    private[uoption] lazy val wrap: WrappedNone =
```

Comment:
Btw, are these required with the class being `private[uoption]`?

Is this in case you forget to hide these if you make the class non-private?